pl-faded-parsons.js

```diff
@@ -523,6 +523,34 @@ class ParsonsWidget {
           },
         }),
       );
+    
+    // add uids to each line and blank for logging ////////////////////////////
+    //  + add logging hooks for blank edits       ////////////////////////////
+    $(widget.config.main)
+      .find(".codeline-tray")
+      .each((trayNumber, tray) =>
+        $(tray)
+          .find("li.codeline")
+          .each((codelineNumber, codeline) => {
+            if ($(codeline).attr("logging-id")) return;
+            const codelineId = `${trayNumber}.${codelineNumber}`;
+            $(codeline).attr("logging-id", codelineId);
+            findBlanksIn(codeline).each((blankNumber, blank) => {
+              $(blank).attr("logging-id", `${codelineId}.${blankNumber}`);
+
+              $(blank).on({ // logging hooks for blank edits
```

Comment:
Recommend pulling this out of the big nested callback, use `findBlanksIn(widget.config.main).on` instead, give it it's own comment description